src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,62 @@
+package duke;
+
+import java.io.IOException;
+
+import java.time.format.DateTimeParseException;
+
+import duke.command.Command;
+import duke.exception.DukeException;
+import duke.parser.Parser;
+import duke.storage.Storage;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+public class Duke {
+    private Ui ui;
+    private TaskList taskList;
+    private Storage storage;
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+
+        try {
+            taskList = TaskList.deserialize(storage.load());
+            ui.sayLoadingSuccess();
+        } catch (DateTimeParseException | IllegalArgumentException | IOException e) {
+            ui.sayLoadingFail();
+            taskList = new TaskList();
+        }
+    }
+
+    public void run() {
+        ui.sayGreet();
+        ui.sayHelp();
+
+        String response = "";
+        boolean isExit = false;
+
+        while (!isExit) {
+            try {
+                response = ui.getUserCommand();
+                Command c = Parser.parse(response);
+                c.execute(taskList, ui, storage);
+                isExit = c.isExit();
+            } catch (IOException e) {
+                ui.sayError(e.getMessage());
+            } catch (DukeException e) {
+                ui.sayError(e.getMessage());
+            }
```

Comment:
Perhaps these two catch statements can be merged into one using ` ||` since they have the same body?

Reply:
That's a good idea. Thanks!